Civi/Funding/EventSubscriber/FundingCase/FundingCasePermissionsCacheClearSubscriber.php

```diff
@@ -0,0 +1,159 @@
+<?php
+/*
+ * Copyright (C) 2023 SYSTOPIA GmbH
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published by
+ *  the Free Software Foundation in version 3.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+declare(strict_types = 1);
+
+namespace Civi\Funding\EventSubscriber\FundingCase;
+
+use Civi\Api4\FundingCaseContactRelation;
+use Civi\Core\Event\PreEvent;
+use Civi\Funding\Database\ChangeSetFactory;
+use Civi\Funding\FundingCase\FundingCasePermissionsCacheManager;
+use Civi\RemoteTools\Api4\Api4Interface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * Clears cached funding case permissions, if an entity is
+ * created/updated/deleted that might affect the permissions.
+ */
+final class FundingCasePermissionsCacheClearSubscriber implements EventSubscriberInterface {
+
+  private Api4Interface $api4;
+
+  private ChangeSetFactory $changeSetFactory;
+
+  private FundingCasePermissionsCacheManager $permissionsCacheManager;
+
+  /**
+   * @inheritDoc
+   */
+  public static function getSubscribedEvents(): array {
+    // Use minimum priority, so we also get possible changes by previous event listeners.
+    return [
+      'hook_civicrm_pre::Individual' => ['preContact', PHP_INT_MIN],
+      'hook_civicrm_pre::Organization' => ['preContact', PHP_INT_MIN],
```

Comment:
Same question: is this missing the `Household` contact sub-type?

Reply:
Fixed.